xarray/namedarray/utils.py

```diff
@@ -80,6 +77,83 @@ def to_0d_object_array(
     return result
 
 
+def is_dict_like(value: Any) -> TypeGuard[Mapping[Any, Any]]:
+    return hasattr(value, "keys") and hasattr(value, "__getitem__")
+
+
+def drop_missing_dims(
+    supplied_dims: Iterable[Hashable],
+    dims: Iterable[Hashable],
+    missing_dims: ErrorOptionsWithWarn,
+) -> Iterable[Hashable]:
+    """Depending on the setting of missing_dims, drop any dimensions from supplied_dims that
+    are not present in dims.
+
+    Parameters
+    ----------
+    supplied_dims : Iterable of Hashable
+    dims : Iterable of Hashable
+    missing_dims : {"raise", "warn", "ignore"}
+    """
+
+    if missing_dims == "raise":
+        supplied_dims_set = {val for val in supplied_dims if val is not ...}
+        invalid = supplied_dims_set - set(dims)
+        if invalid:
+            raise ValueError(
+                f"Dimensions {invalid} do not exist. Expected one or more of {dims}"
+            )
+
+        return supplied_dims
+
+    elif missing_dims == "warn":
+        invalid = set(supplied_dims) - set(dims)
+        if invalid:
+            warnings.warn(
+                f"Dimensions {invalid} do not exist. Expected one or more of {dims}"
+            )
+
+        return [val for val in supplied_dims if val in dims or val is ...]
+
+    elif missing_dims == "ignore":
+        return [val for val in supplied_dims if val in dims or val is ...]
+
+    else:
+        raise ValueError(
+            f"Unrecognised option {missing_dims} for missing_dims argument"
+        )
+
+
+def infix_dims(
+    dims_supplied: Collection[Any],
```

Comment:
New typing style again! Try to use _Dims instead.

Reply:
copied these as is -;)... fixing them shortly